modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java

```diff
@@ -499,6 +505,64 @@ public void testQueueMultipleThreads() throws Exception {
 
     }
 
+    /**
+     * Test for ExceptionListener functionality.
+     *
+     * @throws Exception If fails.
+     */
+    public void testExceptionListener() throws Exception {
+        // restart broker with auth plugin
+        if (broker.isStarted()) {
+            broker.stop();
+
+            broker.waitUntilStopped();
+        } else
+            broker.waitUntilStopped();
+
+        broker.setPlugins(new BrokerPlugin[]{new SimpleAuthenticationPlugin(new ArrayList())});
+
+        broker.start(true);
+
+        connFactory = new ActiveMQConnectionFactory(BrokerRegistry.getInstance().findFirst().getVmConnectorURI());
+
+        List<Throwable> lsnrExceptions = new LinkedList<>();
+
+        Destination dest = new ActiveMQQueue(QUEUE_NAME);
+
+        try (IgniteDataStreamer<String, String> dataStreamer = grid().dataStreamer(DEFAULT_CACHE_NAME)) {
+            JmsStreamer<ObjectMessage, String, String> jmsStreamer = newJmsStreamer(ObjectMessage.class, dataStreamer);
+
+            jmsStreamer.setExceptionListener(new ExceptionListener() {
+                @Override public void onException(JMSException e) {
+                    System.out.println("ERROR");
+
+                    lsnrExceptions.add(e);
+                }
+            });
+
+            jmsStreamer.setDestination(dest);
+
+            try {
+                jmsStreamer.start();
+
+                fail("Expecting security exception");
+            } catch (IgniteException ie) {
+                assertTrue("Caught IgniteException, expected: ", ie.getCause().getCause() instanceof SecurityException);
```

Comment:
The message here is printed when the assertion fails, not passes, so it usually reads like "Unexpected exception...".

Reply:
Replaced with assertThrowsWithCause